sql/src/main/java/io/crate/execution/engine/ExecutionPhasesTask.java

```diff
@@ -196,10 +196,12 @@ private void setupContext(Map<String, Collection<NodeOperation>> operationByServ
         List<Tuple<ExecutionPhase, RowConsumer>> handlerPhaseAndReceiver = createHandlerPhaseAndReceivers(
             handlerPhases, handlerConsumers, initializationTracker);
 
+        SharedShardContexts sharedShardContexts = new SharedShardContexts(indicesService, enableProfiling);
         JobExecutionContext.Builder builder = jobContextService.newBuilder(jobId, localNodeId, operationByServer.keySet())
-            .enableProfiling(enableProfiling);
+            .enableProfiling(enableProfiling)
+            .sharedShardContexts(sharedShardContexts);
```

Comment:
I think this shouldn't be the `sharedShardContexts` directly but some kind of `ProfilingContext` (which is also passed into the `SharedShardContexts` - replacing the `enableProfiling` option)

This way the remaining functionality of the `sharedShardContexts` doesn't become available to the `JobExecutionContext`